src/python-common/ceph/deployment/service_spec.py

```diff
@@ -710,16 +702,6 @@ def rgw_frontends_config_value(self) -> str:
             ports.append(f"port={self.get_port()}")
         return f'beast {" ".join(ports)}'
 
-    def validate(self) -> None:
-        super(RGWSpec, self).validate()
-
-        if not self.rgw_realm:
```

Comment:
I think this check now is the opposite..... 
If the user sets a realm and a zone ... It is worth to check if the realm and the zone exists ? 
What if the user only specifies the realm or the zone .... should we use "default" for the missing piece?